entity-service/src/test/kotlin/com/egm/stellio/entity/repository/Neo4jSearchRepositoryTests.kt

```diff
@@ -104,6 +108,158 @@ class Neo4jSearchRepositoryTests : WithNeo4jContainer {
+        createRelationship(EntitySubjectNode(userEntity.id), AUTH_REL_CAN_READ, thirdEntity.id)
+        createRelationship(EntitySubjectNode(userEntity.id), AUTH_REL_CAN_READ, fourthEntity.id)
+
+        val entities = searchRepository.getEntities(
+            QueryParams(
+                types = setOf("Beekeeper"),
+                q = "name==\"Scalpa\"",
+                offset = offset,
+                limit = limit,
+                geoQuery = GeoQuery("near;maxDistance==1500", "Point", "[56.71, 12.79]", NGSILD_LOCATION_PROPERTY)
+            ),
+            sub,
+            DEFAULT_CONTEXTS
+        ).second
+
+        assertTrue(entities.size == 1)
+        assertTrue(entities.containsAll(listOf(firstEntity.id)))
+    }
+
+    @Test
+    fun `it should return matching entities with geo query and minDistance`() {
+        val userEntity = createEntity(userUri, listOf(USER_TYPE), mutableListOf())
+        val firstEntity = createEntity(
+            beekeeperUri,
+            listOf("Beekeeper"),
+            mutableListOf(Property(name = expandedNameProperty, value = "Scalpa")),
+            "POINT (56.7 12.78)"
+        )
+        val secondEntity = createEntity(
+            "urn:ngsi-ld:Beekeeper:1231".toUri(),
+            listOf("Beekeeper"),
+            mutableListOf(Property(name = expandedNameProperty, value = "Scalpa")),
+            "POLYGON ((7.49 43.78, 7.5 43.78, 7.5 43.79, 7.49 43.79, 7.49 43.78))"
+        )
+        val thirdEntity = createEntity(
+            "urn:ngsi-ld:Beekeeper:1232".toUri(),
+            listOf("Beekeeper"),
+            mutableListOf(Property(name = expandedNameProperty, value = "Scalpa"))
+        )
+
+        val fourthEntity = createEntity(
+            "urn:ngsi-ld:Beekeeper:1233".toUri(),
+            listOf("Beekeeper"),
+            mutableListOf(Property(name = expandedNameProperty, value = "Scalpa")),
+            "POINT (56.6 12.78)"
+        )
+
+        createRelationship(EntitySubjectNode(userEntity.id), AUTH_REL_CAN_WRITE, firstEntity.id)
+        createRelationship(EntitySubjectNode(userEntity.id), AUTH_REL_CAN_ADMIN, secondEntity.id)
+        createRelationship(EntitySubjectNode(userEntity.id), AUTH_REL_CAN_READ, thirdEntity.id)
+        createRelationship(EntitySubjectNode(userEntity.id), AUTH_REL_CAN_READ, fourthEntity.id)
+
+        val entities = searchRepository.getEntities(
+            QueryParams(
+                types = setOf("Beekeeper"),
+                q = "name==\"Scalpa\"",
+                offset = offset,
+                limit = limit,
+                geoQuery = GeoQuery("near;minDistance==1500", "Point", "[56.71, 12.79]", NGSILD_LOCATION_PROPERTY)
+            ),
+            sub,
+            DEFAULT_CONTEXTS
+        ).second
+
+        assertTrue(entities.size == 1)
+        assertTrue(entities.containsAll(listOf(fourthEntity.id)))
+    }
+
+    @Test
+    fun `it should return matching entities with geo query and egalDistance`() {
```

Comment:
I'm really not sure that this "equal distance" thing exists. Where did you find it?

Reply:
Not in the spec but when we do `extractGeorelParams` we put `=`. So I treat it. Maybe we put finally `Triple(georel, null, null)` instead `Triple(DISTANCE_QUERY_CLAUSE, "=", comparisonParams[1])`

Reply:
ideally, this should be rejected (as is said in the comment in the `extractGeorelParams` function). Leave as it is for now, but don't write a test for something that should not work (while inventing an `equalDistance` param)